src/Search/Filters/Tags.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace P4\MasterTheme\Search\Filters;
+
+use WP_Term;
+
+/**
+ * Tags used for search.
+ */
+class Tags {
+	/**
+	 * @return WP_Term[]
+	 */
+	public static function get_all(): array {
+		$tags = get_terms(
+			[
+				'taxonomy'   => 'post_tag',
+				'hide_empty' => false,
+			]
+		);
+
+		if ( is_wp_error( $tags )
+			|| ! is_array( $tags )
```

Comment:
If the args are defined here statically in code, it's probably not needed to do the check for whether it's an array. WP should already reliably return an array. It's only a string if you pass in an argument to count instead of fetch.

Because of our args we can probably also assume it's always going to be a `WP_Term` if `is_wp_error()` returns false.

In case it returns a non empty array that has other stuff than `WP_Term`, I guess we would rather like to throw an error, instead of continuing with an empty array?